cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -116,24 +116,38 @@ public void getData(HttpRequest request, HttpResponder responder,
   }
 
   @POST
-  @Path("/previews/{preview-id}/tracers/")
+  @Path("/previews/{preview-id}/tracers")
   public void getTracersData(HttpRequest request, HttpResponder responder,
                              @PathParam("namespace-id") String namespaceId,
                              @PathParam("preview-id") String previewId) throws Exception {
     NamespaceId namespace = new NamespaceId(namespaceId);
     ApplicationId application = namespace.app(previewId);
-    Map<String, List<String>> previewRequest = parseBody(request,
-                                                         new TypeToken<Map<String, List<String>>>() { }.getType());
-    if (previewRequest == null) {
-      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Tracer name is not provided.");
-      return;
-    }
-    Map<String, Map<String, List<JsonElement>>> result = new HashMap<>();
-    for (List<String> tracerNames : previewRequest.values()) {
-      for (String tracerName : tracerNames) {
-        result.put(tracerName, previewManager.getRunner(application).getData(tracerName));
+    try {
+      Map<String, List<String>> previewRequest = parseBody(request,
+                                                           new TypeToken<Map<String, List<String>>>() {
+                                                           }.getType());
+      if (previewRequest == null) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "The body is not provided.");
+        return;
       }
+      Map<String, Map<String, List<JsonElement>>> result = new HashMap<>();
+      for (List<String> tracerNames : previewRequest.values()) {
```

Comment:
its not a good API if you allow arbitrary keys and only use the values. What is the point of the keys then?  I think you should make the list of tracers be for the "tracers" key, which allows you to expand the api in the future if needed to have other arguments as other fields in the request.